internal/signer/darwin/keychain/keychain.go

```diff
@@ -458,19 +498,20 @@ func (k *Key) getRSAEncryptAlgor() (C.SecKeyAlgorithm, error) {
 			algorithms = rsaPKCS1v15Algorithms
 		}
 	default:
-		return 0, fmt.Errorf("unsupported algorithm %T", pub)
+		return UNKNOWN_SECKEY_ALGORITHM, fmt.Errorf("unsupported algorithm %T", pub)
 	}
-	return algorithms[k.hashFunc], nil
+	return algorithms[k.hash], nil
 }
 
 func (k *Key) getEncryptAlgorithm() (C.SecKeyAlgorithm, error) {
-	if k.hashFunc == 0 {
-		k.hashFunc = crypto.SHA256
+	if k.hash == 0 {
+		k.hash = crypto.SHA256
 	}
-	return k.getRSAEncryptAlgor()
+	return k.getRSAEncryptAlgorithm()
 }
 
-func (k *Key) getRSADecryptAlgor() (C.SecKeyAlgorithm, error) {
+func (k *Key) getRSADecryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	UNKNOWN_SECKEY_ALGORITHM := C.CFStringRef(0)
```

Comment:
Make `UNKNOWN_SECKEY_ALGORITHM` a Global and re-use it in all your functions. It's main purpose is to avoid magic numbers. https://en.wikipedia.org/wiki/Magic_number_(programming)